pydatalab/pydatalab/apps/xrd/blocks.py

```diff
@@ -39,11 +39,35 @@ def load_pattern(
         if ext == ".xrdml":
             df = parse_xrdml(location)
 
-        elif ext == ".xy":
-            df = pd.read_csv(location, sep=r"\s+", names=["twotheta", "intensity"])
-
         else:
-            df = pd.read_csv(location, sep=r"\s+", names=["twotheta", "intensity", "error"])
+            if ext == ".xye":
+                columns = ["twotheta", "intensity", "error"]
+            else:
+                columns = ["twotheta", "intensity"]
+
+            # Try to parse the file by incrementing skiprows until all lines can be cast to np.float64
+            skiprows: int = 0
+            # Set arbitrary limit to avoid infinite loop; a header of 10,000 lines is unlikely to be useful
+            while skiprows < 10_000:
+                try:
+                    df = pd.read_csv(
+                        location, sep=r"\s+", names=columns, dtype=np.float64, skiprows=skiprows
```

Comment:
Would it be easier to just use `comment="'"`?  

Annoyingly, some xy/xye-type files might use other comment characters like "#", though I don't believe any of the types we currently support do that...

Reply:
That's the problem, the comment has to be a single char rather than a list. I thought about some other ways of processing this manually but I think this is the most robust generally -- also lets us extract the header as potential metadata though I admit I don't like the hardcoded limit of 10k

Reply:
What about that idea you had of reading the files backwards?

Reply:
I thought about trying to add it to pandas directly (via C) but I think its overkill, and way less efficient than this way if the header is small